File: BirdsApi/Data/Observation.cs
using System.Text.Json.Serialization;
using NetTopologySuite.Geometries;

namespace BirdsApi.Data;

public class Observation
{
    protected Observation() {} // For EF Core

    public Observation(Bird bird)
    {
        Bird = bird;
        ObservationDate = DateTime.Now;
    }

    public int Id { get; private set; }

    public Bird Bird { get; set; }

    public DateTimeOffset ObservationDate { get; set; }

    public string? Remark { get; set; }

    public Point? Location { get; set; }

    [JsonIgnore]
    public Photo? Photo { get; set; }

    public bool HasPhoto => Photo != null;
}

File: BirdsApp/src/app/observations/observation.ts
import { Point } from "geojson";
import { Bird } from "../birds/bird";

export interface Observation {
  id: number;
  bird: Bird;
  observationDate: Date;
  location: Point | undefined;
  remark: string | undefined;
  photo: File | undefined;
  hasPhoto: boolean;
}
